test/libyul/yulOptimizerTests/fullSuite/create2_and_mask.yul

```diff
@@ -0,0 +1,21 @@
+{
+    // This does not optimize the masks away. Due to the way the expression simplifier
+    // is built, it would have to create another `create2` opcode for the simplification
+    // which wolud be fatal.
```

Comment:
What do you mean by that?

Reply:
The problem is that the expression simplifier can only work if the code is in ssa form:
```
let a := create2(...)
let b := and(a, 0xfff...)
```

The correct replacement would be
```
let b := create2(...)
```

but for that, it is crucial that `let a := create2(...)` is removed and this is not something the CSE can do in its current design.